File: cvdl2020/hw2/dev/tool.js
const cv2 = require("opencv4nodejs");
/**
 * 
 * @param {cv2.VideoCapture} capture
 * @param {number} colorCode
 */
const getFrames = (capture, colorCode) => {
    /**
     * @type {cv2.Mat[]}
     */
    let frames = []
    while (1) {
        let frame = capture.read()
        if (frame.empty) {
            break
        }
        if (colorCode !== undefined) frame = frame.cvtColor(colorCode)
        frames.push(frame)
    }
    return frames
}
/**
 * 
 * @param {cv2.VideoCapture} capture
 * @param {number} colorCode
 */
const getFramesBuffer = (capture, colorCode) => {
    let frame = capture.read().cvtColor(colorCode)
    /**
     * @type {Buffer}
     */
    let framesBuffer = frame.getData()
    while (1) {
        frame = capture.read()
        if (frame.empty) {
            break
        }
        if (colorCode !== undefined) frame = frame.cvtColor(colorCode)
        framesBuffer = Buffer.concat([framesBuffer, frame.getData()])
    }
    return framesBuffer
}

module.exports = { getFrames, getFramesBuffer }

File: cvdl2020/hw2/py/tool.py
from cv2 import cv2 as cv2


def get_frames(capture: cv2.VideoCapture, colorCode: int = None):
    frames = []
    while (True):
        ret, frame = capture.read()
        if not ret:
            break
        if colorCode != None:
            frame = cv2.cvtColor(frame, colorCode)
        frames.append(frame)
    return frames


def get_frames_buffer(capture: cv2.VideoCapture, colorCode: int = None):
    frames = get_frames(capture, colorCode)
    buffer = frames[0].data.tobytes()
    for frame in frames[1:]:
        buffer += frame.data.tobytes()
    return buffer
